### ISSUE TITLE
FixedWindowRollingPolicy exceeds configured sizes

### ISSUE THREAD

**Issue by shimonmagal** [tomcat-slf4j-logback/tomcat-slf4j-logback#122]
Hi, @hazendaz 

When using FixedWindowRollingPolicy with straight forward configuration,
it sometimes writes over than the configured size files (20MB in my specific configuration [true for bigger sizes as well]), and sometimes I see that catalina.1.log (which isn't active file) is still being written to.

I've made sure no other process or webapp access this log. Also, tested the same logback version (1.1.7) with other plain java / webapp reporting massive amount of logs, and they all rolled over around 20.0 MB or 20.1 MB at most.

This once almost filled my disk.

Is easily produceble with 1MB and setting log level to DEBUG. Version: 8.5.6-with-logback-1.1.7

The configuration:

 <appender name="FILE-catalina-main" class="org.apache.juli.logging.ch.qos.logback.core.rolling.RollingFileAppender">
    <File>${MY_HOME}/logs/tomcat/catalina.log</File>
    <rollingPolicy class="org.apache.juli.logging.ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
      <FileNamePattern>${MY_HOME}/logs/tomcat/catalina.%i.log</FileNamePattern>
      <MinIndex>1</MinIndex>
      <MaxIndex>10</MaxIndex>
    </rollingPolicy>
    
    <triggeringPolicy class="org.apache.juli.logging.ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
      <MaxFileSize>20MB</MaxFileSize>
    </triggeringPolicy>
    
    <encoder>
      <pattern>[%d{dd-MM HH:mm:ss.SSS}][-------TOMCAT------] %-5level - %msg {%logger{0}} [%thread] %n</pattern>
    </encoder>
  </appender>

**Comment by hazendaz** [comment]
@shimonmagal I have not seen this issue before.  There really isn't much special about our configuration other than it basically tricks tomcat into using slf4j/logback.  So I suspect it should work exactly the same as logback.  I don't employ log cycling like that in a production environment.  Frankly I wouldn't advice that anyways as it overwrites older logs.  If you use more of the approach we setup, it's going to keep making backups which can be controlled over time periods and might be more enterprise ready for something like splunk than using that style.  You might have to keep digging on this and if you find the issue please submit a pull request.  I'll leave this open for now.

**Comment by hazendaz** [comment]
@shimonmagal There were some bugs somewhat related to this that logback team corrected in lasted release.  The next release I put out will be on that version.  It fixed a time based issue similar to what you describe.  When I do get to that point (waiting on tomcat 6 final release), possibly can you then provide feedback if it still exists?  This might be a few more weeks from now.

**Comment by hazendaz** [comment]
Closing issue due to no response.  I believe this is related to problems in slf4j which recently have been fixed.  Given we have upgraded quite a lot since, if this issue is real, others would be raising it and/or feedback would need to come in.  In an effort to keep old possibly now inaccurate items off our issues list, closing.  

@shimonmagal If you test master and still believe this is a problem, please open a new issue or re-open this one.
